rajawali/src/main/java/org/rajawali3d/renderer/RajawaliPipRenderer.java

```diff
@@ -0,0 +1,192 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package org.rajawali3d.renderer;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.MotionEvent;
+
+import org.rajawali3d.materials.Material;
+import org.rajawali3d.materials.textures.ATexture;
+import org.rajawali3d.renderer.pip.SubRenderer;
+import org.rajawali3d.renderer.pip.WorkaroundScreenQuad;
+import org.rajawali3d.scene.RajawaliScene;
+
+/**
+ * <p>
+ * Creates a renderer that renders two scenes through separate <code>SubRenderer</code>s.
+ * In order to use this, first create the <code>PipRenderer</code>, then create and assign the
+ * <code>SubRenderer</code>s through <code>setMainRenderer</code> (for the main content)
+ * <code>setMinirenderer</code> (for the content in the mini view).
+ * The <code>pipScale</code> and <code>pipMargin</code> parameters control how the mini scene
+ * is placed.
+ * </p>
+ */
+public class RajawaliPipRenderer extends RajawaliRenderer {
+    private static final String TAG = "RajawaliPipRenderer";
+
+    private RenderTarget mMainRenderTarget;
+    private RenderTarget mMiniRenderTarget;
+
+    private WorkaroundScreenQuad mMiniQuad;
+    private WorkaroundScreenQuad mMainQuad;
+    private Material mMiniQuadMaterial;
+    private Material mMainQuadMaterial;
+
+    private RajawaliScene mCompositeScene;
+
+    private SubRenderer mMiniRenderer;
+    private SubRenderer mMainRenderer;
+
+    // These variables control where the minimap is placed. Note they are specified in standardized
+    // OpenGL coordinates 0 to 1
+    private final float pipScale;
+    private final float pipMarginX;
+    private final float pipMarginY;
+
+    private float miniXmin, miniXmax, miniYmin, miniYmax;
+
+    /**
+     * @param pipScale   Size of the mini view from 0 to 1, i.e.: ratio of full screen to mini
+     * @param pipMarginX Margin space from the mini view to the edge of the main view in pixels
+     * @param pipMarginY Margin space from the mini view to the edge of the main view in pixels
+     */
+    public RajawaliPipRenderer(Context context, float pipScale, float pipMarginX, float pipMarginY) {
+        super(context);
+        this.pipScale = pipScale;
```

Comment:
Its kind of a personal preference thing, but I very much dislike the use of the 'this' pointer, unless passing this object to another method. Since pipScale and the like are private final fields, they should be declared as mPipScale, etc which will eliminate this need.